When both the content file and .treeinfo file are missing from the media
look in the media.1/products and media.1/media files for information.
Caasp 4.0 has not content or .treeinfo file on the media

Index: virt-manager-3.3.0/virtinst/install/urldetect.py
===================================================================
--- virt-manager-3.3.0.orig/virtinst/install/urldetect.py
+++ virt-manager-3.3.0/virtinst/install/urldetect.py
@@ -505,8 +505,23 @@ class _SuseDistro(_RHELDistro):
             cache.checked_for_suse_content = True
             content_str = cache.acquire_file_content("content")
             if content_str is None:
-                return False
-
+                products_str = cache.acquire_file_content("media.1/products")
+                if products_str:
+                    products_str = products_str.replace('/', ' ,', 1)
+                    products_str = "DISTRO " + products_str.replace('-', ' ')
+                media_str = cache.acquire_file_content("media.1/media")
+                if media_str:
+                    media_arch = "x86_64"
+                    if 'aarch64' in media_str:
+                        media_arch = "aarch64"
+                    elif 'ppc64le' in media_str:
+                        media_arch = "ppc64le"
+                    elif 's390x' in media_str:
+                        media_arch = "s390x"
+                    media_str = 'BASEARCHS ' + media_arch
+                if products_str is None and media_str is None:
+                    return False
+                content_str = products_str + "\n" + media_str
             try:
                 cache.suse_content = _SUSEContent(content_str)
             except Exception as e:  # pragma: no cover
